### PR TITLE
CUSTCOM-253 Adapt AJAX test for script reloading

### PR DESCRIPTION
**Pull request by MattGill98**
This module had no actual test, so it's been adapted to test for
reloading a script which contains a null EL value.

### CODE REVIEW

File: jsf/ajax/src/test/java/org/javaee7/jsf/ajax/AjaxScriptTest.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    public void setup() throws IOException {
+        try (WebClient webClient = new WebClient()) {
+            // Make sure to wait for AJAX requests in the foreground
+            webClient.setAjaxController(new NicelyResynchronizingAjaxController());
+            page = webClient.getPage(base + "/faces/index.xhtml");
+        }
+    }
+
+    @Test
+    public void when_ajax_button_clicked_expect_no_errors() throws IOException, InterruptedException {
```
[the remark this answers is not in the document]
Comment (MattGill98) [reply jsf/ajax/src/test/java/org/javaee7/jsf/ajax/AjaxScriptTest.java:46]: Most tests are just `testSomething`, which I didn't find to be particularly descriptive when running a bunch of tests. IMO it should be clear from the method name what went wrong in the tests, especially when running such a large test suite

**Comment by MattGill98** [comment]
Hmm the test fails against 5.201? I'll run it against master and see if it needs editing...

**Comment by MattGill98** [comment]
Seems to fail, can you try again?

File: jsf/ajax/src/test/java/org/javaee7/jsf/ajax/AjaxScriptTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (dmatej) [reply jsf/ajax/src/test/java/org/javaee7/jsf/ajax/AjaxScriptTest.java:46]: The content is alright, I only prefer "camelCase" instead of "not_camel_case"

**Comment by dmatej** [comment]
```
mvn clean verify -Ppayara-server-managed -Dpayara.version=5.201 -pl :jsf -amd
...
Unpacking /home/dmatej/.m2/repository/fish/payara/distributions/payara/5.201/payara-5.201.zip to /home/dmatej/work/repo/git/javaee7-samples/target with includes "" and excludes ""
...
```
image

**Comment by dmatej** [comment]
Do we use same profiles? Maybe I missed something ...

**Comment by dmatej** [comment]
Ok, I got it, I had to run mvn clean before it, because the Payara is unpacked to the root directory, so clean with -pl does not remove it and unzip does not remove existing files. Now it failed with 5.201 :)

**Pull request merged**